src/transaction.ts

```diff
@@ -33,7 +33,8 @@ export function craftSingleRecipientPset(
 
   const fee = createFeeOutput(
     firstSelection.selectedUtxos.length,
-    1 + 1,
+    // Change outputs + 1 recipient + 1 fee
```

Comment:
the fee output is already counted in. Looks like to me that extra 1 was the change of the recipients, which should be already included in `changeOutputs`

Reply:
also, it can happen that in the second round of coinselection it adds more inputs  & changes.

Maybe is worth updating the fee output in that case? cc/ @louisinger

Reply:
`estimateTxSize` in `createFeeOutput` takes numOutputs. I don't see where fee output is already counted in.

Reply:
> also, it can happen that in the second round of coinselection it adds more inputs & changes.

for a single recipient tx (with 1 asset), second round of coinselection shouldn't add an additional changes (either there is a change or not). So either a single recipient has 2 outputs or 3 outputs. I don't see any other cases. Anyway, yes this should be computed after the final coin selection.

Reply:
yep exactly non-lbtc asset send is the issue

Reply:
changeOutputs + 2 looks good to me. However I noticed that the `estimateTxSize` assumes the tx is using segwit everywhere. It may be possible that we need to take into account segwit or not to compute the size of the transaction.

Reply:
For I think its fine for this PR then, will open ticket for this